sdk/clientcore/core/src/main/java/io/clientcore/core/utils/CompositeChallengeHandler.java

```diff
@@ -50,8 +50,7 @@ public void handleChallenge(HttpRequest request, Response<BinaryData> response,
         }
 
         // Log an error if no handler could handle the challenge
-        LOGGER.throwableAtError()
-            .log("None of the challenge handlers could handle the challenge.", UnsupportedOperationException::new);
+        LOGGER.atError().log("None of the challenge handlers could handle the challenge.");
```

Comment:
This changed from throwing an exception to just logging. Do we not want to throw here?

Reply:
we didn't throw here and there is a test that validates that we didn't throw, so I guess it's intentional.
@alzimmermsft could you confirm that it's the right behavior not to throw here?

Reply:
Yeah, this was intentional, but if we don't like the design I'm happy to change it.

The reasoning behind this is that we could configure a challenge handle but receive a call where there's nothing to handle.